scripts/play.ts

```diff
@@ -0,0 +1,380 @@
+ * `appendRandomHash` to virtually guarantee uniqueness.
+ *
+ * @see {@link appendRandomHash}
+ */
+function createRandomName() {
+  const ADJECTIVES = [
+    "adorable",
+    "beautiful",
+    "charming",
+    "cloudy",
+    "delightful",
+    "focused",
+    "honest",
+    "inspiring",
+    "lovable",
+    "melodious",
+    "pleasant",
+    "pretty",
+    "shiny",
+    "sugary",
+    "tall",
+    "valliant",
+  ];
+
+  const PASTRIES = [
+    "baguette",
+    "briouat",
+    "crepe",
+    "croissant",
+    "donut",
+    "eclair",
+    "flaons",
+    "kolompeh",
+    "macaron",
+    "paste",
+    "pretzel",
+    "rustico",
+    "semla",
+    "strudel",
+    "taiyaki",
+    "toast",
+  ];
+
+  const adjective = ADJECTIVES[Math.floor(Math.random() * ADJECTIVES.length)];
+  const pastry = PASTRIES[Math.floor(Math.random() * PASTRIES.length)];
+
+  return appendRandomHash(`${adjective}-${pastry}`);
+}
+
+/**
+ * Appends a random 7 digit hash to a string.
+ *
+ * @param input - The string to which the hash is appended.
+ */
+function appendRandomHash(input: string) {
+  const hash = crypto
+    .createHash("sha1")
+    .update(crypto.randomUUID())
+    .digest("hex")
+    .toString()
+    .slice(-7);
+
+  return `${input}-${hash}`;
+}
+
+/**
+ * Returns the name of the last modified file or directory in a given directory.
+ *
+ * @param dir - The directory to search.
+ * @param options - Options that determine the function's behavior.
+ */
+async function getLastModifiedEntry(
+  dir: string,
+  options?: {
+    /**
+     * A list of file or directory names to exclude.
+     */
+    exclude?: string[];
+
+    /**
+     * Determines if only directories are considered.
+     */
+    dirOnly?: boolean;
+  },
+): Promise<string | undefined> {
+  let entries = await fs.readdir(dir, { withFileTypes: true });
+
+  if (options?.exclude) {
+    entries = entries.filter((entry) => !options.exclude?.includes(entry.name));
+  }
+
+  if (options?.dirOnly) {
+    entries = entries.filter((entry) => entry.isDirectory());
+  }
+
+  if (entries.length < 1) {
+    return;
+  }
+
+  return entries.sort((a, b) => {
+    return (
+      fsSync.statSync(path.join(b.path, b.name)).mtimeMs -
+      fsSync.statSync(path.join(a.path, a.name)).mtimeMs
+    );
+  })[0].name;
+}
+
+async function pathExists(path: string): Promise<boolean> {
+  try {
+    await fs.access(path);
+
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function updateJSONFile<TSchema extends Record<string, unknown>>(
+  filePath: string,
+  contents: Partial<TSchema>,
+  options?: {
+    /**
+     * If `true`, commands are not executed.
+     */
+    dryRun?: boolean;
+  },
+) {
+  if (options?.dryRun) {
+    console.info(`Update ${filePath} with ${JSON.stringify(contents)}`);
+
+    return;
+  }
+
+  const existingContents = JSON.parse(await fs.readFile(filePath, "utf8"));
+  const newContents = deepMerge(existingContents, contents);
```

Comment:
I opted for a deep merge function since I would like to provide partial values to `updateJSONFile`. A deep merge is necessary if the partial value replaces an object, like `scripts`.

We could probably simplify the `updatePackageJSON` and `updateSliceMachineConfig` functions (wrappers around `updateJSONFile` introduced after your review) since we know exactly which properties we want to edit. For example, we could manually merge `scripts`. What do you think?